db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/ReferencedKeyConstraintDescriptor.java

```diff
@@ -270,9 +263,8 @@ else if (type == DISABLED)
 	 *
 	 * @return false
 	 */
-	public boolean isReferenced()
-	{
-		return referenceCount != 0;
+	public boolean isReferenced() throws StandardException {
+		return getReferenceCount() > 0;
```

Comment:
The optimization in `getForeignKeyConstraints()` no longer applies, because for the "fast" path we are getting the list of foreign keys from the data dictionary. It actually hurts because if we have the list cached we will always request it from the DD first. Please remove the optimization there and coordinate with @hatyo to make sure this doesn't make the creation of large FK hierarchies too slow.

Reply:
Thinking a bit more about it, we could use the same cache field `fkConstraintList` in `getReferenceCount()`, if it's already populated we have the list and don't need to retrieve it from the DD. If we haven't, we retrieve the list and update the field.